src/components/Dashboard/Charts/SentEmailCampaigns/SentEmailCampaigns.tsx

```diff
@@ -1,76 +1,119 @@
+
+type OnPopoverOpen = (message: Message) => (...args: Parameters<ReturnType<typeof usePopover>['onOpen']>) => void
+
+interface MessageListProps {
+  onPopoverOpen: OnPopoverOpen
+  messages: Message[]
 }
 
-const SentEmailCampaigns = ({ isMailsStatutory = false }) => {
-  const { handleError } = useErrorHandler()
-  const popover = usePopover()
-  const currentMessage = useRef<Message | null>(null)
-  const queryKey = useScopedQueryKey(['paginated-campaigns', { feature: 'Dashboard', view: 'SentEmailCampaigns' }])
-  const onPopoverOpen = useCallback(
-    (message: Message) =>
-      (...args: Parameters<typeof popover.onOpen>) => {
-        currentMessage.current = message
-        return popover.onOpen(...args)
-      },
-    [popover]
-  )
+const MessageList = (props: MessageListProps) => (
+  <AGrid container spacing={3}>
+    <TransitionGroup component={null}>
+      {props.messages.map(message => (
+        <CSSTransition key={message.id} timeout={500} classNames="item">
+          <Grid item xs={12} sm={6} md={4} xl={3} data-cy="email-campaign-card">
+            <MessageCard message={message} onPopoverOpen={props.onPopoverOpen} />
+          </Grid>
+        </CSSTransition>
+      ))}
+    </TransitionGroup>
+  </AGrid>
+)
 
-  const {
-    data: paginatedCampaigns = null,
-    fetchNextPage,
-    hasNextPage,
-    isLoading,
-  } = useInfiniteQuery<{ data: Message[] }>(
-    queryKey,
-    ({ pageParam: page = 1 }) => getMessages({ isMailsStatutory, page }),
-    {
-      getNextPageParam,
-      onError: handleError,
-    }
-  )
+const MessageLoader = () => (
+  <Stack justifyContent="center" alignItems="center" height="10vh">
+    <CircularProgress />
+  </Stack>
+)
 
-  const campaigns = usePaginatedData(paginatedCampaigns) as Message[]
+const EmptyStatus = (props: { text: string }) => (
+  <Stack justifyContent="center" alignItems="center" height="10vh">
+    <Typography variant="subtitle2" textAlign="center">
+      {props.text}
+    </Typography>
+  </Stack>
+)
 
-  const drafts = campaigns.filter(message => message.draft)
-  const sents = campaigns.filter(message => !message.draft)
+interface SearchBoxProps {
+  onPopoverOpen: OnPopoverOpen
+}
 
-  if (isLoading) {
-    return <Loader isCenter />
-  }
-  if (!campaigns.length) {
-    return <div>{isMailsStatutory ? messages.noStatutoryMail : messages.noCampaign}</div>
+const SearchBox = (props: SearchBoxProps) => {
+  const [search, setSearch] = useState('')
+  const debouncedSearch = useDebounce(search, 500)
+  const queryKey = useScopedQueryKey([
+    'search-campaigns',
+    { feature: 'Dashboard', view: 'SentEmailCampaigns' },
+    debouncedSearch,
+  ])
+  const { handleError } = useErrorHandler()
+  const { data, isFetching } = useQuery({
+    queryFn: () => getMessages({ pagination: false, label: search }),
+    queryKey: queryKey,
+    onError: handleError,
+    enabled: debouncedSearch.trim().length > 2,
```

Comment:
peut-on appliquer trim avant de remplire le state et éviter de le refaire ici ? 
```suggestion
    enabled: debouncedSearch.length > 2,
```